doc/sf/number_series.qbk

```diff
@@ -259,6 +259,67 @@ Passing a value greater than `max_prime` results in a __domain_error being raise
 
 This function is `constexpr` only if the compiler supports C++14 constexpr functions.
 
+[endsect] [/section:primes Prime Numbers]
+
+[section:primes Prime Sieve]
+
+[h4 Synopsis]
+
+``
+#include <boost/math/special_functions/prime_sieve.hpp>
+``
+
+namespace boost { namespace math {
+
+    template<class ExecutionPolicy, typename Z, class OutputIterator>
+    auto prime_sieve(ExecutionPolicy&& policy, Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<class ExecutionPolicy, class Z, class OutputIterator>
+    auto prime_range(ExecutionPolicy&& policy, Z lower_bound, Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<typename Z, class OutputIterator>
+    auto prime_sieve(Z upper_bound, OutputIterator output) -> decltype(output)
+
+    template<class Z, class OutputIterator>
+    auto prime_range(Z lower_bound, Z upper_bound, OutputIterator output) -> decltype(output)
+
+
+}} // End namespaces
+
+[h4 Description]
+
+There are two sets of functions available `prime_sieve` and `prime_range`. `prime_sieve` will return all primes in the
+range 2 to `upper_bound` inclusive. `prime_range` will return all the primes in the range `lower_bound` to `upper_bound`
+inclusive.
+
+If you have a C++17 compatible compiler you are able to pass an execution policy to both functions. Any
+policy besides `std::execution::seq` will enable internal multi-threading. If your compiler is not C++17 compatible the
+sequential overloads will be used.
+
+For upper_bound <= 2^24 two threads will be used. For any value larger than 2^24 `std::thread::hardware_concurrency()` will be called,
+and all available concurrency will be used. Additionally, the memory requirements are `O(sqrt(N))`.
+
+/Nota bene:/ If `std::thread::hardware_concurrency()` returns 0 the max number of threads will be set to 2.
+
+Example:
+    // To calculate primes 2 - 1,000,000 in parallel
+    std::vector<int64_t> primes;
+    boost::math::prime_sieve(std::execution::par, 1000000, std::back_inserter(primes));
+
+    // To calculate primes 100 - 1,000 sequentially
+    std::vector<int64_t> primes;
+    boost::math::prime_range(100, 1000, std::back_inserter(primes));
+
+
+/Nota bene:/ For values larger than 2^24 the returned primes will not be sorted
```

Comment:
I don't understand; why not sort them?

Or: Why not leave all of them unsorted?

Reply:
Bad logic merging the lists during parallel execution. It is fixed now.